astropy/coordinates/builtin_frames/galactocentric.py

```diff
@@ -82,100 +102,231 @@ class galactocentric_frame_defaults(ScienceState):
+    @classmethod
+    def get_solar_params_from_string(cls, arg):
+        """
+        Return Galactocentric solar parameters given string names
+        for the parameter sets.
+
+        Returns
+        -------
+        parameters : dict
+            Copy of Galactocentric solar parameters from registry
+
+        Raises
+        ------
+        KeyError
+            If invalid string input to registry
+            to retrieve solar parameters for Galactocentric frame.
+
+        """
+        return cls.get_from_registry(arg)["parameters"]
 
     @classmethod
     def validate(cls, value):
         if value is None:
             value = cls._latest_value
 
         if isinstance(value, str):
-            params, refs = cls.get_solar_params_from_string(value)
-            cls._references = refs
-            return params
+            state = cls.get_from_registry(value)
+            cls._references = state["references"]
+            cls._state = state
+            parameters = state["parameters"]
 
         elif isinstance(value, dict):
-            return value
+            parameters = value
 
         elif isinstance(value, Galactocentric):
             # turn the frame instance into a dict of frame attributes
-            attrs = dict()
+            parameters = dict()
             for k in value.frame_attributes:
-                attrs[k] = getattr(value, k)
-            cls._references = value.frame_attribute_references()
-            return attrs
+                parameters[k] = getattr(value, k)
+            cls._references = value.frame_attribute_references.copy()
+            cls._state = dict(parameters=parameters,
+                              references=cls._references)
 
         else:
             raise ValueError("Invalid input to retrieve solar parameters for "
                              "Galactocentric frame: input must be a string, "
                              "dict, or Galactocentric instance")
 
+        return parameters
+
+    @classmethod
+    def register(cls, name: str, parameters: dict, references=None,
+                 **meta: dict):
+        """Register a set of parameters.
+
+        Parameters
+        ----------
+        name : str
+            The registration name for the parameter and metadata set.
+        parameters : dict
+            The solar parameters for Galactocentric frame.
+        references : dict or None, optional
+            References for contents of `parameters`.
+            None becomes empty dict.
+        **meta: dict, optional
+            Any other properties to register.
+
+        """
+        # check on contents of `parameters`
+        must_have = ("galcen_coord", "galcen_distance", "galcen_v_sun",
+                     "z_sun", "roll")
+        missing = [k for k in must_have if k not in parameters]
+        if missing:
+            raise ValueError(f"Missing parameters: {missing}")
+
+        state = dict(parameters=parameters, references=references)
```

Comment:
If we change `references=None` to `{}` here, can we not get rid of the extra checks on `references` since now we know the `state` is OK?

Reply:
Done. and extraneous checks are now gone.